### DISCUSSION TITLE
Text transformations

### DISCUSSION THREAD

**Discussion by xo-o** [fabricjs/fabric.js#9690] [Show and tell]
Fabric text transformations experiment.

Link [here](https://fabric-text-transformation.vercel.app/)

![preview](https://github.com/fabricjs/fabric.js/assets/149122294/ecdc8601-2db0-419d-8769-8e11d42013fe)

**Comment by ShaMan123** [comment]
Wow! Amazing work
We are planning to rewrite text, it would be great to have this or perhaps as a standalone package
Is it open source?
How is performance when the text editable?

**Reply by ShaMan123** [reply]
I see that dblclick allows editing the text without the transform
Good enough I guess though I wonder how perf would be to execute the transform on each key change

**Reply by ShaMan123** [reply]
This is  a long lasting desired feature BTW

**Reply by ShaMan123** [reply]
Did you use WebGL?

**Reply by xo-o** [reply]
No WebGL. Paper.js and custom controls does the job.

**Reply by ShaMan123** [reply]
Did you use fabric?

**Reply by xo-o** [reply]
> No WebGL. Paper.js and custom controls does the job.

I mean, fabric custom controls. v5 and v6-beta.

**Reply by ShaMan123** [reply]
Cool
I wonder how easy it is to do using WebGL
